### PR TITLE
feat: allow usage of already unzipped electron versions

### PR DESCRIPTION
**Pull request by aryanshridhar**
Link to discussion - https://github.com/electron/fiddle/pull/1134#discussion_r952949487

### CODE REVIEW

File: src/installer.ts [hunk 1]
```diff
@@ -297,24 +337,36 @@ export class Installer extends EventEmitter {
...
-      // @ts-ignore: yes, I know noAsar isn't defined in process
-      const { noAsar } = process;
-      try {
-        // @ts-ignore: yes, I know noAsar isn't defined in process
-        process.noAsar = true;
-        await extract(zipFile, { dir: electronInstall });
-      } finally {
-        // @ts-ignore: yes, I know noAsar isn't defined in process
-        process.noAsar = noAsar; // eslint-disable-line
+      const { path: zipFile, preInstalled: isAlreadyInstalled } =
```
Comment (erickzhao) [inline src/installer.ts:340]: nit(non-blocking): it's a bit confusing that `isAlreadyInstalled` is in a different control flow branch from the `already installed` debug statement.

Would `alreadyExtracted` be clearer?

File: tests/installer.test.ts [hunk 2]
```diff
@@ -113,13 +119,29 @@ describe('Installer', () => {
...
+    return { events, binaryConfig };
+  }
+
+  async function unZipBinary(dirName = '13.1.7'): Promise<string> {
+    const extractDir = path.join(tmpdir, 'downloads', dirName);
+    fs.mkdirSync(path.join(tmpdir, 'downloads', dirName), {
+      recursive: true,
+    });
+
+    await extract(fixture('electron-v13.1.7.zip'), {
```
Comment (erickzhao) [inline tests/installer.test.ts:140]: question: why are we hard-coding `electron-v13.1.7.zip` in the test but making the `dirName` configurable?

File: tests/installer.test.ts [hunk 3]
```diff
@@ -113,13 +119,29 @@ describe('Installer', () => {
...
+    const { path: zipfile } = binaryConfig;
 
-    expect(isDownloaded).toBe(true);
-    expect(fs.existsSync(zipfile)).toBe(true);
-    expect(installer.state(version)).toBe(downloaded);
+    if (safe) {
+      expect(isDownloaded).toBe(true);
+      expect(fs.existsSync(zipfile)).toBe(true);
+      expect(installer.state(version)).toBe(downloaded);
+    }
```
Comment (erickzhao) [inline tests/installer.test.ts:129]: question: What exactly does the `safe` parameter mean here?

File: src/installer.ts (hunk 1 above)
Comment (aryanshridhar) [reply src/installer.ts:340]: Yup, But now that I think of - maybe `alreadyExtracted` goes better in the `ElectronBinary` interface than `preInstalled`.

File: tests/installer.test.ts (hunk 2 above)
Comment (aryanshridhar) [reply tests/installer.test.ts:140]: If I understand if correctly, I was using that for testing and debugging. Removed it now.

**Review by erickzhao — APPROVED** [review]
Aside: it makes it easier for reviewers if you don't force push between rounds of feedback since you can see the diff between rounds a lot easier. We usually squash merge PRs anyways so the amount of commits in a pull request doesn't really matter!

**Pull request merged**